gazebo_plugins/src/gazebo_ros_imu_sensor.cpp

```diff
@@ -0,0 +1,229 @@
+/* Copyright [2015] [Alessandro Settimi]
+ * 
+ * email: ale.settimi@gmail.com
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.*/
+
+#include <gazebo_plugins/gazebo_ros_imu_sensor.h>
+#include <iostream>
+#include <gazebo/sensors/ImuSensor.hh>
+#include <gazebo/physics/World.hh>
+
+GZ_REGISTER_SENSOR_PLUGIN(gazebo::GazeboRosImuSensor)
+
+gazebo::GazeboRosImuSensor::GazeboRosImuSensor(): SensorPlugin()
+{
+  accelerometer_data = math::Vector3(0, 0, 0);
+  gyroscope_data = math::Vector3(0, 0, 0);
+  orientation = math::Quaternion(1,0,0,0);
+  seed=0;
+}
+
+void gazebo::GazeboRosImuSensor::Load(gazebo::sensors::SensorPtr sensor_, sdf::ElementPtr sdf_)
+{
+  sdf=sdf_;
+  sensor=dynamic_cast<gazebo::sensors::ImuSensor*>(sensor_.get());
```

Comment:
You should check against null pointer after the dynamic_cast here. Maybe the plugin was loaded for a sensor which is not an `ImuSensor`.